src/wheredataframe/wheredataframe.jl

```diff
@@ -0,0 +1,287 @@
+"""
+    WhereDataFrame{<:AbstractDataFrame,<:AbstractIndex}
+
+    The result of a [`where`](@ref) operation on an `AbstractDataFrame`; a
+    subset of a `AbstractDataFrame`
```

Comment:
can you please highlight that this is a wrapper (as opposed to `SubDataFrame` which is `AbstractDataFrame`?)

Reply:
Yes. Just to be clear, this draft is just a way to start a discussion about this potential syntax — I figured it may be interesting for people to try it out a bit.

Reply:
Understood - I highlighted in the comments the major design considerations I see with this proposal.